android/KMAPro/build.sh

```diff
@@ -1,39 +1,75 @@
 #!/usr/bin/env bash
-# Build KMAPro
-
-# Set sensible script defaults:
-# set -e: Terminate script if a command returns an error
-set -e
-# set -u: Terminate script if an unset variable is used
-set -u
-# set -x: Debugging use, print each statement
+# Build Keyman for Android app (KMAPro)
+
 # set -x
+set -eu
 
 ## START STANDARD BUILD SCRIPT INCLUDE
 # adjust relative paths as necessary
 THIS_SCRIPT="$(readlink -f "${BASH_SOURCE[0]}")"
 . "${THIS_SCRIPT%/*}/../../resources/build/build-utils.sh"
 ## END STANDARD BUILD SCRIPT INCLUDE
 
+. "$KEYMAN_ROOT/resources/shellHelperFunctions.sh"
+
 . "$KEYMAN_ROOT/resources/build/build-download-resources.sh"
 
-echo Build KMAPro
+# This script runs from its own folder
+cd "$THIS_SCRIPT_PATH"
+
+# ################################ Main script ################################
+
+# Definition of global compile constants
+
+CONFIG="release"
+BUILD_FLAGS="build -x lint -x test"                     # Gradle build w/o test
+TEST_FLAGS="-x assembleRelease lintRelease testRelease" # Gradle test w/o build
+DAEMON_FLAG=
+
+builder_describe "Builds Keyman for Android app." \
+  "@../KMEA" \
+  "clean" \
+  "configure" \
+  "build" \
+  "test             Runs lint and unit tests." \
+  "--ci             Don't start the Gradle daemon. For CI" \
+  "--upload-sentry  Upload to sentry"
+
+# parse before describe_outputs to check debug flags
+builder_parse "$@"
+
+if builder_has_option --debug; then
+  builder_heading "### Debug config ####"
+  CONFIG="debug"
+  BUILD_FLAGS="assembleDebug -x lint -x test"
+  TEST_FLAGS="-x assembleDebug lintDebug testDebug"
+fi
+
+builder_describe_outputs \
+  configure     kMAPro/libs/keyman-engine.aar \
+  build:app     kMAPro/build/outputs/apk/$CONFIG/keyman-${VERSION}.apk
+
+#### Build
+
+function _convert_markdown_to_html() {
```

Comment:
```suggestion
function convert_markdown_to_html() {
```

No need to use underscores here. In the builder.inc.sh that delineation is necessary to separate internal functions from public functions.